src/core/fetchers/manifest/manifest_fetcher.ts

```diff
@@ -0,0 +1,288 @@
+ * ).subscribe(({ value }) => {
+ *   console.log("Manifest:", value.manifest);
+ * });
+ * ```
+ */
+export default class ManifestFetcher {
+  private _backoffOptions : IBackoffOptions;
+  private _manifestUrl : string | undefined;
+  private _pipelines : ITransportManifestPipeline;
+
+  /**
+   * @param {string | undefined} url
+   * @param {Object} pipelines
+   * @param {Object} backoffOptions
+   */
+  constructor(
+    url : string | undefined,
+    pipelines : ITransportPipelines,
+    backoffOptions : IManifestFetcherBackoffOptions
+  ) {
+    this._manifestUrl = url;
+    this._pipelines = pipelines.manifest;
+    this._backoffOptions = getManifestBackoffOptions(backoffOptions);
+  }
+
+  /**
+   * (re-)Load the Manifest without yet parsing it.
+   *
+   * You can set an `url` on which that Manifest will be requested.
+   * If not set, the regular Manifest url - defined on the
+   * `ManifestFetcher` instanciation - will be used instead.
+   * @param {string} [url]
+   * @returns {Observable}
+   */
+  public fetch(url? : string) : Observable<IManifestFetcherResponse |
+                                   IManifestFetcherWarningEvent> {
+    const requestUrl = url ?? this._manifestUrl;
+
+    // TODO Remove the resolver completely in the next major version
+    const resolver : IManifestResolverFunction =
+      this._pipelines.resolver ??
+      // TODO Wrong implem seems to be considered by the linter here.
+      // Is this a tslint bug?
+      /* tslint:disable deprecation */
+      observableOf;
+      /* tslint:enable deprecation */
+
+    const loader : IManifestLoaderFunction = this._pipelines.loader;
+
+    return tryCatch$(resolver, { url: requestUrl }).pipe(
+      catchError((error : Error) : Observable<never> => {
+        throw errorSelector(error);
+      }),
+      mergeMap((loaderArgument : IManifestLoaderArguments) => {
+        const loader$ = tryCatch$(loader, loaderArgument);
+        return tryRequestObservableWithBackoff(loader$, this._backoffOptions).pipe(
+          catchError((error : unknown) : Observable<never> => {
+            throw errorSelector(error);
+          }),
+          map((evt) => {
+            return evt.type === "retry" ?
+              ({ type: "warning" as const, value: errorSelector(evt.value) }) :
+              ({ type: "response" as const,
+                 parse: (parserOptions : IManifestFetcherParserOptions) => {
+                   return this._parseLoadedManifest(evt.value.value, parserOptions);
+                 } });
+          }));
+      }));
+  }
+
+  /**
+   * Parse an already loaded Manifest.
+   *
+   * This method should be reserved for Manifests for which no request has been
+   * done.
+   * In other cases, it's preferable to go through the `fetch` method, so
+   * information on the request can be used by the parsing process.
+   * @param {*} manifest
+   * @param {Object} parserOptions
+   * @returns {Observable}
+   */
+  public parse(
```

Comment:
Why should we have a public and private function to parse the manifest ?
The fetcher could call the public parser direclty, no ?

Reply:
To make the `ManifestFetcher` API easier to understand for external code.

From the outside there is only two methods:
  - `fetch(url)` which fetch and parse the Manifest stored at `url`
  - `parse(manifest, parserOptions)` which only parses an already-fetched Manifest.

Here the API is very simple, we can just know what do to just by looking at the function signatures.
The issue is that in reality, the real private method actually calling the parsing code rely on metadata from the Manifest request (like the size of the loaded data and the duration of the request).

The `fetch` method has it internally (as the request was done in it) so it can provide them easily.
When parsing directly the Manifest with no request however, no such data is neither available nor make sense.

So instead of making a single public method with added arguments that made no sense when the method was called directly, I went to a readable public method with a reduced API and a private method with more arguments that actually do the parsing,

Reply:
Clear !